### PR TITLE
Instrumentation support for AWS SDK v2

### PR DESCRIPTION
**Pull request by sanojsubran**
This PR adds support for the Instana instrumentation of [version 2 of the AWS SDK for Go](https://github.com/aws/aws-sdk-go-v2).

For details on the changes related to AWS SDK V2, refer [here](https://aws.github.io/aws-sdk-go-v2/docs/).

The approach has been relatively same with respect to the previous version of our instrumentation SDK. The middleware stack of AWS SDK has changed from version 1 and version 2. Hence I added the relevant changes for version 2. The previous middleware stack was simpler but may be it had some limitations. The latest one offers lots of configurability options.

Another point, I tried to change is to limit the number of public APIs of our instrumentation package and provide better test cases which will not need many private APIs.

### CODE REVIEW

**Review by willianpc — COMMENTED** [review]
Pls check

File: instrumentation/instaawsv2/aws.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	case instana.FieldS:
+		c.Attrs.Set(fieldS, val)
+	case instana.FieldL:
+		c.Attrs.Set(fieldL, val)
+	}
+}
+
+func (c messageAttributesCarrier) ForeachKey(handler func(key, val string) error) error {
+	if v, ok := c.Attrs.Get(fieldT); ok {
+		handler(instana.FieldT, v)
```
[the remark this answers is not in the document]
Comment (sanojsubran) [reply instrumentation/instaawsv2/aws.go]: Added the suggested changes.
I need to revisit this file again. Currently the implementation is taken from v1 instrumentation. I see scope for improvement.

**Review by willianpc — APPROVED** [review]
Good job!
As per our conversation, the work looks pretty good.
I leave it to you if you want to consider my proposal or not, since those are mostly related to personal preference than improvements to the code.

**Comment by sanojsubran** [comment]
> Good job! As per our conversation, the work looks pretty good. I leave it to you if you want to consider my proposal or not, since those are mostly related to personal preference than improvements to the code.

I think we can merge the changes. I agree that it looks cleaner.

**Pull request merged**